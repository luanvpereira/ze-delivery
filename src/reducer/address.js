import { combineReducers } from 'redux';

import { CHANGE_CURRENT_ADDRESS } from '../types';

export const currentAddress = (state = null, { type, payload }) => {
	switch (type) {
		case CHANGE_CURRENT_ADDRESS:
			return payload;

		default:
			return state;
	}
};

export default combineReducers({
	currentAddress
});
