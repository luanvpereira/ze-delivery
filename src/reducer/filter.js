import { combineReducers } from 'redux';

import {
	CHANGE_METHOD_ADDRESS,
	CHANGE_CURRENT_CATEGORY,
	CHANGE_FILTER_TEXT
} from '../types';

export const id = (state = null, { type, payload }) => {
	switch (type) {
		case CHANGE_METHOD_ADDRESS:
			return payload;

		default:
			return state;
	}
};

export const categoryId = (state = 0, { type, payload }) => {
	switch (type) {
		case CHANGE_CURRENT_CATEGORY:
			return payload;

		default:
			return state;
	}
};

export const search = (state = '', { type, payload }) => {
	switch (type) {
		case CHANGE_FILTER_TEXT:
			return payload;

		default:
			return state;
	}
};

export default combineReducers({
	id,
	categoryId,
	search
});
