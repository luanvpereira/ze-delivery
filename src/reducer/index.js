import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

export const middlewares = () => [thunk];

export const initializeStore = (initialState = {}) =>
	createStore(
		combineReducers({}),
		initialState,
		composeWithDevTools(applyMiddleware(...middlewares())),
	);
