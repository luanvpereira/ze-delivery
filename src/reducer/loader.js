import {
	SHOW_LOADER,
	HIDE_LOADER
} from '../types';

export default (state = false, { type }) => {
	switch (type) {
		case SHOW_LOADER:
			return true;

		case HIDE_LOADER:
			return false;

		default:
			return state;
	}
};
