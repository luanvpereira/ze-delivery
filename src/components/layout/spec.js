import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import { initializeStore } from '../../reducer';

import Layout from './';

describe('<Layout/>', () => {
	it('should render a <Layout/>', () => {
		const wrapper = mount(
			<Provider store={initializeStore({
				loader: true
			})}>
				<Layout>
					<p>Corpo</p>
				</Layout>
			</Provider>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
