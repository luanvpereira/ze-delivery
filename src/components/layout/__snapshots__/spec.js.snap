// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> should render a <Layout/> 1`] = `
<Layout>
  <div
    className="layout"
  >
    <Header>
      <header
        className="header"
      >
        <Container>
          <div
            className="container"
          >
            <h1
              className="headerTitle"
            >
              <Link
                href="/"
              >
                <a
                  className="headerLogo"
                  href="/"
                  onClick={[Function]}
                >
                  Ze Delivery
                </a>
              </Link>
            </h1>
             
          </div>
        </Container>
      </header>
    </Header>
    <p>
      Corpo
    </p>
    <Footer>
      <footer
        className="footer"
      >
        <Container>
          <div
            className="container"
          >
            <p
              className="footerCopyright"
            >
              © 
              2019
               Zé Delivery - Todos os direitos reservados.
            </p>
            <p
              className="footerLogo"
            >
              Ze Delivery
            </p>
            <ul
              className="footerList"
            >
              <li>
                <a
                  className="footerListItem"
                  href="https://zedelivery.zendesk.com/hc/pt-br"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Dúvidas?
                </a>
              </li>
              <li>
                <a
                  className="footerListItem"
                  href="/terms"
                >
                  Termos de Condições e Política de 
                  <br />
                   Privacidade
                </a>
              </li>
              <li>
                <a
                  className="footerListItem"
                  href="/partner"
                >
                  Seja nosso parceiro
                </a>
              </li>
            </ul>
             
          </div>
        </Container>
      </footer>
    </Footer>
  </div>
</Layout>
`;
